x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -144,23 +189,86 @@ function DatatableComponent(props: DatatableProps & { formatFactory: FormatFacto
         columns={props.args.columns.columnIds
           .map(field => {
             const col = firstTable.columns.find(c => c.id === field);
+            const colIndex = firstTable.columns.findIndex(c => c.id === field);
+
+            const filterable = col?.meta?.type && props.getType(col.meta.type)?.type === 'buckets';
             return {
               field,
               name: (col && col.name) || '',
+              render: (value: unknown) => {
+                const formattedValue = formatters[field]?.convert(value);
+                if (filterable && value) {
+                  return (
+                    <EuiFlexGroup
+                      className="lnsDataTable__cell"
+                      data-test-subj="lnsDataTableCellValueFilterable"
+                      gutterSize="xs"
+                    >
+                      <EuiFlexItem grow={false}>{formattedValue}</EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiFlexGroup
+                          responsive={false}
+                          gutterSize="none"
+                          alignItems="center"
+                          className="lnsDataTable__filter"
+                        >
+                          <EuiToolTip
+                            position="bottom"
+                            content={
+                              <FormattedMessage
+                                id="xpack.lens.filterForValueButtonTooltip"
+                                defaultMessage="Filter for value"
+                              />
```

Comment:
Nitpick: I'm not sure we need to use a `FormattedMessage` here, as the tooltip has a default font size. The FormattedMessage appears larger than necessary to me.